pocld/daemon.cc

```diff
@@ -302,6 +556,14 @@ int PoclDaemon::launch(std::string ListenAddress, struct ServerPorts &Ports,
     POCL_MSG_ERR("Error resolving address: %s\n", gai_strerror(error));
     return -1;
   }
+
+#ifdef ENABLE_REMOTE_ADVERTISEMENT_AVAHI
+  avahiAdvertiseP = StartAvahiAdvert(ResolvedAddress, Ports);
```

Comment:
Not necessary to do in this PR now, but you *could* use std::unique_ptr since nothing else ever touches this pointer. Then you can assign it here with `variablename.reset(StartAvahiAdvert(ResolvedAddress, Ports));` and don't need to do anything manually in the PoclDaemon destructor.

Reply:
Ah yes, that's cool!